test/org/sagebionetworks/bridge/TestUserAdminHelper.java

```diff
@@ -85,40 +89,45 @@ public StudyIdentifier getStudyIdentifier() {
         }
     }
     @Autowired
-    public void setUserAdminService(UserAdminService userAdminService) {
+    public final void setUserAdminService(UserAdminService userAdminService) {
         this.userAdminService = userAdminService;
     }
     @Autowired
-    public void setAuthService(AuthenticationService authService) {
+    public final void setAuthService(AuthenticationService authService) {
         this.authService = authService;
     }
     @Autowired
-    public void setStudyService(StudyService studyService) {
+    public final void setStudyService(StudyService studyService) {
         this.studyService = studyService;
     }
 
     public String makeRandomUserName(Class<?> cls) {
+        checkNotNull(cls);
         String devPart = BridgeConfigFactory.getConfig().getUser();
         String rndPart = TestUtils.randomName(cls);
         return String.format("bridge-testing+%s-%s", devPart, rndPart);
     }
 
     public TestUser createUser(Class<?> cls) {
-        return createUser(cls, null);
+        return createUser(cls, true, true, null, null);
     }
-
-    public TestUser createUser(Class<?> cls, Set<Roles> roles) {
-        checkNotNull(cls, "Class must not be null");
-
-        return createUser(cls, true, true, roles);
+    
+    public TestUser createUser(Class<?> cls, Study study, boolean consent) {
+        return createUser(cls, study, true, consent, null, null);
     }
 
-    public TestUser createUser(Class<?> cls, boolean signIn, boolean consent, Set<Roles> roles) {
-        checkNotNull(cls, "Class must not be null");
-
-        String name = makeRandomUserName(cls);
-        SignUp signUp = new SignUp(name, name + "@sagebase.org", PASSWORD, roles);
+    public TestUser createUser(Class<?> cls, boolean signIn, boolean consent) {
+        return createUser(cls, signIn, consent, null, null);
+    }
+    
+    public TestUser createUser(Class<?> cls, boolean signIn, boolean consent, Set<Roles> roles, Set<String> dataGroups) {
         Study study = studyService.getStudy(TEST_STUDY_IDENTIFIER);
+        return createUser(cls, study, signIn, consent, roles, dataGroups);
+    }
+    
+    public TestUser createUser(Class<?> cls, Study study, boolean signIn, boolean consent, Set<Roles> roles, Set<String> dataGroups) {
+        String name = makeRandomUserName(cls);
+        SignUp signUp = new SignUp(name, name + EMAIL_DOMAIN, PASSWORD, roles, dataGroups);
         return createUser(signUp, study, signIn, consent);
     }
```

Comment:
Instead of having an every increasing number of "convenience" overrides, would it be cleaner to just use a builder? How much refactoring would this require?